server/channels/jobs/export_users_to_csv/export_users_to_csv.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package export_users_to_csv
+
+import (
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/v8/channels/jobs"
+	"github.com/mattermost/mattermost/server/v8/channels/store"
+	"github.com/pkg/errors"
+)
+
+const (
+	timeBetweenBatches = 1 * time.Second
+)
+
+type ExportUsersToCSVAppIFace interface {
+	jobs.BatchReportWorkerAppIFace
+	GetUsersForReporting(filter *model.UserReportOptions) ([]*model.UserReport, *model.AppError)
+}
+
+// MakeWorker creates a batch migration worker to delete empty drafts.
+func MakeWorker(jobServer *jobs.JobServer, store store.Store, app ExportUsersToCSVAppIFace) model.Worker {
+	return jobs.MakeBatchReportWorker(
+		jobServer,
+		store,
+		app,
+		timeBetweenBatches,
+		"csv",
+		model.UserReport.GetHeaders(model.UserReport{}), // TODO no?
+		getData,
+	)
+}
+
+// parseJobMetadata parses the opaque job metadata to return the information needed to decide which
+// batch to process next.
+func parseJobMetadata(data model.StringMap) (*model.UserReportOptions, error) {
+	options := model.UserReportOptionsAPI{
+		UserReportOptionsWithoutDateRange: model.UserReportOptionsWithoutDateRange{
+			SortColumn:          "Username",
+			PageSize:            100,
+			LastSortColumnValue: data["last_column_value"],
+			LastUserId:          data["last_user_id"],
+		},
+		DateRange: data["date_range"],
+	}
+
+	return options.ToBaseOptions(time.Now()), nil
+}
+
+// makeJobMetadata encodes the information needed to decide which batch to process next back into
+// the opaque job metadata.
+func makeJobMetadata(lastColumnValue string, userID string) model.StringMap {
+	data := make(model.StringMap)
+	data["last_column_value"] = lastColumnValue
+	data["last_user_id"] = userID
+
+	return data
+}
+
+func getData(jobData model.StringMap, app ExportUsersToCSVAppIFace) ([]model.ReportableObject, model.StringMap, bool, error) {
```

Comment:
This method is called in a loop, with specified job data saved in the DB. It will pull a page of users from the DB and send it out to be saved. Then it will increment the page and continue looping.